Penumbra/Import/Models/Import/VertexAttribute.cs

```diff
@@ -121,89 +122,174 @@ public static VertexAttribute Position(Accessors accessors, IEnumerable<Accessor
             element,
             index => {
-                // Blend weights are _very_ sensitive to float imprecision - a vertex sum being off
-                // by one, such as 256, is enough to cause a visible defect. To avoid this, we tweak
-                // the converted values to have the expected sum, preferencing values with minimal differences.
-                var originalValues = values[index];
-                var byteValues = BuildNByte4(originalValues);
-
-                var adjustment = 255 - byteValues.Select(value => (int)value).Sum();
-                while (adjustment != 0)
-                {
-                    var convertedValues = byteValues.Select(value => value * (1f / 255f)).ToArray();
-                    var closestIndex = Enumerable.Range(0, 4)
-                        .Where(index => {
-                            var byteValue = byteValues[index];
-                            if (adjustment < 0) return byteValue > 0;
-                            if (adjustment > 0) return byteValue < 255;
-                            return true;
-                        })
-                        .Select(index => (index, delta: Math.Abs(originalValues[index] - convertedValues[index])))
-                        .MinBy(x => x.delta)
-                        .index;
-                    byteValues[closestIndex] = (byte)(byteValues[closestIndex] + Math.CopySign(1, adjustment));
-                    adjustment = 255 - byteValues.Select(value => (int)value).Sum();
-                }
-                
-                return byteValues;
+                var weight0       = weights0[index];
+                var weight1       = weights1?[index];
+                var originalData  = BuildUshort4(weight0, weight1 ?? Vector4.Zero);
+                var byteValues    = originalData.Select(x => (byte)Math.Round(x * 255f)).ToArray();
+                return AdjustByteArray(byteValues, originalData);
             }
         );
+       
+        /*var element = new MdlStructs.VertexElement()
+        {
+            Stream = 0,
+            Type   = (byte)MdlFile.VertexType.NByte4,
+            Usage  = (byte)MdlFile.VertexUsage.BlendWeights,
+        };
+
+        var weights0 = weights0Accessor.AsVector4Array();
+
+        return new VertexAttribute(
+            element,
+            index =>
+            {
+                var weight0 = weights0[index];
+                var originalData   = new[] { weight0.X, weight0.Y, weight0.Z, weight0.W };
+                var byteValues     = originalData.Select(x => (byte)Math.Round(x * 255f)).ToArray();
+                var newByteValues = AdjustByteArray(byteValues, originalData);
+                if (!newByteValues.SequenceEqual(byteValues))
+                    notifier.Warning("Adjusted blend weights to maintain precision.");
+                return newByteValues;
+            });*/
+    }
+
+    private static byte[] AdjustByteArray(byte[] byteValues, float[] originalValues)

```

Comment:
given this is a blendweight-specific problem (and the `byteValues` and `originalValues` are somewhat implementation details), would probably be cleaner to make this `BuildBlendWeights` with the vec4s as args and return the finalised byte array directly.

Reply:
let me know if the changes are aligned with what you meant here